rohmu/object_storage/base.py

```diff
@@ -202,7 +205,31 @@ def copy_file(
         cannot be copied with this method. If no metadata is given copies the existing metadata."""
         raise NotImplementedError
 
-    def copy_files_from(self, *, source: BaseTransfer[SourceStorageModelT], keys: Collection[str]) -> None:
+    def copy_files_from(
+        self,
+        *,
+        source: BaseTransfer[SourceStorageModelT],
```

Comment:
nit: having a generic appear only once in a function signature doesn't really do anything?  I am not convinced that this has an advantage over `BaseTransfer[Any]`.